### ISSUE TITLE
KeyError: Conv2d( ...  while using custom resnet18 and High Level Pruning example from GIT repo

### ISSUE THREAD

**Issue by LeroyGitBash** [VainF/Torch-Pruning#80]
Hey, 
im using the oficial pruning exampgle from the GIT repo. Instead of using an densenet im unsing an resnet18 witch was pretrained by miselve for image classification. 

wehn running the line pruner.step() 
```
ori_size = tp.utils.count_params(model_ft)
    example_inputs = torch.randn(1, 3, 224, 224)
    imp = tp.importance.MagnitudeImportance(p=2) # L2 norm pruning
    ignored_layers = []
    for m in model_ft.modules():
        if isinstance(m, torch.nn.Linear) and m.out_features == 1000:
            ignored_layers.append(m)
    
    total_steps = 5 
    pruner = tp.pruner.LocalMagnitudePruner( 
        model_ft,
        example_inputs,
        importance=imp,
        total_steps=total_steps, # number of iterations
        ch_sparsity=0.5, # channel sparsity
        ignored_layers=ignored_layers, # ignored_layers will not be pruned
    )
    
    for i in range(total_steps): # iterative pruning
        pruner.step() 
```

im getting the following error 

```
Traceback (most recent call last):
  File "C:/Programs/PyCharmInst/plugins/python-ce/helpers/pydev/pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Programs\PyCharmInst\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:\Tools\Leroy Freyler\20_Projekte\NNCompression\Python\NN_Train\prune_resnet18.py", line 272, in <module>
    pruner.step()
  File "c:\programs\gitpkg\torchpruning\torch_pruning\pruner\basepruner.py", line 114, in step
    for plan in self.get_all_plans():
  File "c:\programs\gitpkg\torchpruning\torch_pruning\pruner\basepruner.py", line 88, in get_all_plans
    m, pruning_fn, list(range(layer_channels))
  File "c:\programs\gitpkg\torchpruning\torch_pruning\dependency.py", line 481, in get_pruning_plan
    root_node = self.module2node[module]
KeyError: Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)
```

Im on the newest version of Torch-Prune installt via "git clone ... "

**Comment by VainF** [comment]
Have you ever tried with the resnet from torchvision? It seems that the layer ``Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)`` was not executed during forwarding and thus was not traced.

**Comment by VainF** [comment]
The following example works for me with branch v1.0

```python
import torch
from torchvision.models import resnet18 as entry
import torch_pruning as tp

model = entry(pretrained=True)
print(model)
# Global metrics
example_inputs = torch.randn(1, 3, 224, 224)
imp = tp.importance.MagnitudeImportance(p=2)
ignored_layers = []
for m in model.modules():
    if isinstance(m, torch.nn.Linear) and m.out_features == 1000:
        ignored_layers.append(m)

iterative_steps = 1
pruner = tp.pruner.MagnitudePruner(
    model,
    example_inputs,
    importance=imp,
    iterative_steps=iterative_steps,
    ch_sparsity=0.5, # remove 50% channels, ResNet18 = {64, 128, 256, 512} => ResNet18_Half = {32, 64, 128, 256}
    ignored_layers=ignored_layers,
)

for i in range(iterative_steps):
    ori_size = tp.utils.count_params(model)
    pruner.step()
    print(model)
    print(model(example_inputs).shape)
    print(
        "  Params: %.2f M => %.2f M"
        % (ori_size / 1e6, tp.utils.count_params(model) / 1e6)
    )
```

**Comment by LeroyGitBash** [comment]
Hello VainF, 

thank you for your quick reply. 

I have been able to locate my error. 

When I trained the model before, I set all parameters except the last layer (fc) to the value "require_grad=False".

Since I am using a global purning technique in your example, this naturally led to violations in the forward and backward hook. 

I was able to quickly fix this error by manually setting all parameters to "requires_grad=True" beforehand. 

Many greetings
Leroy
